### ISSUE TITLE
Duotone using theme/Style colors does not update when theme/Style colors are changed

### ISSUE THREAD

**Issue by jordesign** [WordPress/gutenberg#49595]
## What I expect

When a duotone filter is using the theme's preset colors (eg. Contrast/Primary/Secondary set from Global Styles) I expect the duotone to update when I make changes to the color presets.

## What happens
Instead when changing the preset colors - the duotone filter retains the old colors and shows them as hex values *instead* of the preset name.

### Step-by-step reproduction instructions

1. Set up an image/cover block with the Duotone Filter. For the highlight and shadow values - select preset colors.
2. Change the preset colors in Site Editor > Styles > Colors
3. Return to the image/cover block and see that the duotone filter has not updated.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/1842363/229946433-4fe65f08-d723-42a8-9520-170ddb222a3c.mov

### Environment info

- WP6.2 & GB15.4
- TwentyTwentyThree

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jordesign** [comment]
For additional context it seems that is related to the Duotone values being stores as hex instead of CSS variables.
Related issue here - but specifically related to Style variations:
https://github.com/WordPress/gutenberg/issues/46547

And more related to the Social Icons block:
https://github.com/WordPress/gutenberg/issues/35480
https://github.com/WordPress/gutenberg/issues/36462

There's also an existing PR for switching to CSS Custom properties for Duotone
https://github.com/WordPress/gutenberg/pull/42870

**Comment by jordesign** [comment]
_more_ additional context....

I ran into this issue when trying to create a Block Pattern for submission. The use case being that it would be great to have a pattern be inserted and have the Duotone _automatically_ reference the user's preset colors when inserted.

**Comment by annezazu** [comment]
@MaggieCabrera and @ajlende curious what you all think here!

**Comment by MaggieCabrera** [comment]
Yeah, I believe it's because are saved as hex, when that gets solved we should check that this issue gets resolved as well
